app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -42,14 +48,97 @@ export const ArtifactDetail = ({
-            ))}
-          </div>
-        ))}
+        {environments.map(({ name, state, deployedAt, replacedAt, replacedBy }) => {
+          const deployedAtMillis = DateTime.fromISO(deployedAt).toMillis();
+          const replacedAtMillis = DateTime.fromISO(replacedAt).toMillis();
+          const { version: replacedByPackageVersion, buildNumber: replacedByBuildNumber } =
+            parseName(replacedBy || '') || {};
+
+          return (
+            <div key={name}>
+              <h3>{name.toUpperCase()}</h3>
+              {state === 'deploying' && (
+                <NoticeCard
+                  className="sp-margin-l-right"
+                  icon="md-actuation-launched"
+                  text={undefined}
+                  title="Deploying"
+                  isActive={true}
+                  noticeType="info"
+                />
+              )}
+              {state === 'current' && deployedAt && (
+                <NoticeCard
+                  className="sp-margin-l-right"
+                  icon="cloud-check"
+                  text={undefined}
+                  title={
+                    <span>
+                      Deployed {relativeTime(deployedAtMillis)}{' '}
+                      <span className="text-italic text-regular sp-margin-xs-left">
+                        ({timestamp(deployedAtMillis)})
+                      </span>
+                    </span>
+                  }
+                  isActive={true}
+                  noticeType="success"
+                />
+              )}
+              {state === 'previous' && (
+                <NoticeCard
+                  className="sp-margin-l-right"
+                  icon="checkbox-indeterminate"
+                  text={undefined}
+                  title={
+                    <span className="sp-group-margin-xs-xaxis">
+                      Decommissioned {relativeTime(replacedAtMillis)}{' '}
+                      <span className="text-italic text-regular sp-margin-xs-left">
+                        ({timestamp(replacedAtMillis)})
+                      </span>{' '}
+                      <span className="text-regular">—</span> <span className="text-regular">replaced by </span>
+                      <Pill
+                        text={
+                          replacedByBuildNumber ? `#${replacedByBuildNumber}` : replacedByPackageVersion || replacedBy
+                        }
+                      />
+                    </span>
+                  }
+                  isActive={true}
+                  noticeType="neutral"
+                />
+              )}
+              {state === 'vetoed' && (
+                <NoticeCard
+                  className="sp-margin-l-right"
+                  icon="skull"
+                  text={undefined}
+                  title={
+                    <span className="sp-group-margin-xs-xaxis">
+                      Marked as bad{' '}
+                      {deployedAt && (
```

Comment:
if the version got manually vetoed before ever making it into an environment, we won't have a `deployedAt`

Reply:
Worth dropping that into a comment?

Reply:
Yeah good call ➕

Reply:
after thinking about it, I actually went one step further and explicitly surfaced a user-facing state for this (rather than only giving people looking at the code that context):

<img width="917" alt="Screen Shot 2020-03-23 at 1 37 14 PM" src="https://user-images.githubusercontent.com/1850998/77361057-c70ff100-6d0b-11ea-8f01-29b59b25cb75.png">